pkg/chunkenc/unordered_test.go

```diff
@@ -141,6 +141,26 @@ func Test_Unordered_InsertRetrieval(t *testing.T) {
 			},
 			dir: logproto.BACKWARD,
 		},
+		{
+			desc: "ts remove exact dupe forward",
+			input: []entry{
+				{0, "a"}, {0, "b"}, {0, "b"}, {1, "c"},
```

Comment:
I think I'm missing something here. Should we have a test that has unordered input to make sure the dedupes are still correct as that's the case this PR is addressing?

Reply:
The exact code being tested currently doesn't really care if it's ordered or not, but I did update the tests to be unordered in the spirit of the PR and maybe to better future proof us, good eye though, appreciate you taking the time to review the tests :)